sttz.InstallUnity/Installer/Platforms/WIndowsPlatform.cs

```diff
@@ -0,0 +1,316 @@
+            return true;
+        }
+
+        public async Task Uninstall(Installation installation, CancellationToken cancellation = default)
+        {
+            var result = await RunAsAdmin(Path.Combine(installation.path, "Editor", "Uninstall.exe"), "/AllUsers /Q /S");
+            if (result.exitCode != 0)
+            {
+                throw new Exception($"Could not uninstall Unity. output: {result.output}, error: {result.error}");
+            }
+
+            Logger.LogDebug($"Unity {installation.version} uninstalled successfully");
+
+            try
+            {
+                // TODO: Should folder be deleted even when uninstall command returns with exitcode != 0?
+                Logger.LogInformation($"Deleting folder path {installation.path} recursively");
+                await Task.Delay(1000); // Wait for uninstallation
+                Directory.Delete(installation.path, true);
+
+                Logger.LogDebug($"Folder path {installation.path} deleted");
+            }
+            catch (UnauthorizedAccessException _)
+            {
+                try
+                {
+                    // Sometimes access to folders and files are still in use by Unity uninstall, so we wait some more
+                    await Task.Delay(3000);
+                    Directory.Delete(installation.path, true);
+
+                    Logger.LogDebug($"Folder path {installation.path} deleted at second attempt");
+                }
+                catch (Exception e)
+                {
+                    Logger.LogError(e, $"Failed to delete folder path {installation.path} at second attempt");
+                    // Continue even though errors occur deleting file path
+                }
+            }
+            catch (Exception e)
+            {
+                Logger.LogError(e, $"Failed to delete folder path {installation.path}");
+                // Continue even though errors occur deleting file path
+            }
+        }
+
+        // -------- Helpers --------
+
+        ILogger Logger = UnityInstaller.CreateLogger<WIndowsPlatform>();
+
+        VersionMetadata installing;
+        string installationPaths;
+        bool installedEditor;
+
+        async Task<(int exitCode, string output, string error)> RunAsAdmin(string filename, string arguments)
+        {
+            var startInfo = new ProcessStartInfo();
+            startInfo.FileName = filename;
+            startInfo.Arguments = arguments;
+            startInfo.CreateNoWindow = true;
+            startInfo.WindowStyle = ProcessWindowStyle.Hidden;
+            startInfo.RedirectStandardError = true;
+            startInfo.RedirectStandardOutput = true;
+            startInfo.UseShellExecute = false;
+            startInfo.WorkingDirectory = Environment.CurrentDirectory;
+            startInfo.Verb = "runas";
+            try
+            {
+                var p = Process.Start(startInfo);
+                p.WaitForExit();
```

Comment:
You can switch this to
```suggestion
                await p.WaitForExitAsync();
```
To make it actually async (new in dotnet 5+)